Tests/SpeziTests/DependenciesTests/DependencyTests.swift

```diff
@@ -245,6 +245,7 @@ final class DependencyTests: XCTestCase { // swiftlint:disable:this type_body_le
         let module3 = TestModule3()
         let spezi = Spezi(standard: DefaultStandard(), modules: [TestModule1(), module3])
 
+        throw XCTSkip("Skipped for now!") // TODO: what the fuck?
```

Comment:
TODO to address this before the merge.

Reply:
I’m not sure how to resolve these other than removing them completely. Enabling Swift 6 will make these crash due to task isolation checks.

as explained here https://github.com/StanfordSpezi/Spezi/pull/117#issuecomment-2383038408

Reply:
I would need to take a deeper look at this; but would there be any path to test these preconditions on a non-`@MainActor` execution basis? Or force that we somehow can inject the actor that things should be executed on and assume that this is not a big deal on a longer-term basis as things will be quickly destroyed on a test-case basis and the code should never really run on a production system anyways?

Reply:
This is where the check happens in the swift runtime https://github.com/swiftlang/swift/blob/10b3ed37f23e92c066c381c58ddc8814a8b986e2/stdlib/public/Concurrency/Actor.cpp#L402

We would need to modify the [implementation of the MainActor](https://github.com/swiftlang/swift/blob/10b3ed37f23e92c066c381c58ddc8814a8b986e2/stdlib/public/Concurrency/MainActor.swift#L20) to bypass this. I’m not quite sure if this is possible.

Reply:
Hmm, that might be tough but we could explore that.

Wondering if there is any other reasonable way to work around this?

Reply:
I think it makes more sense to explore ways to refactor these cases. I think, if preconditions are part of the behavior, it might make more sense to replace them with proper error handling. We still can choose to transform an error that was thrown on a lower level to a precondition failure on a top level (e.g., prevent an app start without a cycle in the dependency chain). However, it would be easily possible to, e.g., make certain parts of the DependencyManager throwing and expose throwing overloads for testing.

I think this fundamentally the easier route. We will need to see how easy this is doable with all of the cases below.

What do you think?

Reply:
Good point; that might also make it more reusable in the future if we ever want to use it beyond the current use cases 🚀

Reply:
I added some explicit errors types to replace some of our precondition failures and updated the tests to use and assert this new error type 🚀